include/jsTypeFactory.hh

```diff
@@ -16,7 +16,26 @@
 #include <Python.h>
 
 
-struct PythonExternalString;
+struct PythonExternalString : public JSExternalStringCallbacks {
+public:
+  /**
+   * @brief Get the PyObject using the given char buffer
+   *
+   * @param chars - the char buffer of the PyObject
+   * @return PyObject* - the PyObject string
+   */
+  static PyObject *getPyString(const char16_t *chars);
+
+  /**
+   * @brief decrefs the underlying PyObject string when the JSString is finalized
+   *
+   * @param chars - The char buffer of the string
+   */
+  void finalize(char16_t *chars) const override;
+
+  size_t sizeOfBuffer(const char16_t *chars, mozilla::MallocSizeOf mallocSizeOf) const override;
```

Comment:
when is this called?

Reply:
```cpp
struct PythonExternalString : public JSExternalStringCallbacks {
```
https://hg.mozilla.org/releases/mozilla-esr115/file/tip/js/public/GCAPI.h#l553

Reply:
then probably the return value should not be 0 below